oak_echo_linux_init/src/init.rs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use nix::mount::{mount, MsFlags};
+use std::{env::set_current_dir, fs::create_dir};
+
+/// Performs the minimum initialization required from the initial process on Linux to allow an
+/// application on an initial RAM disk to operate as expected.
+pub fn init() -> anyhow::Result<()> {
+    set_current_dir("/")?;
+
+    // Create mount points.
+    create_dir("/proc")?;
```

Comment:
Are these directories created by this binary? Or by Linux on startup?

Reply:
Yes, these are created by this binary. The Linux kernel does not create them. In a normal Linux distribution there is an initialisation process that will create them, but since we are booting only the kernel we have to do that work.